pylabnet/hardware/nanopositioners/smaract.py

```diff
@@ -0,0 +1,370 @@
+        self._nanopositionersdll.SA_CTL_Close.restype = ctypes.c_uint32 # result status
+
+        # Getting device properties
+        self._nanopositionersdll.SA_CTL_GetProperty_i32.argtypes = [
+            ctypes.c_uint32,                # device handle
+            ctypes.c_int8,                  # index of addressed device, module, or channel
+            ctypes.c_uint32,                # property key
+            ctypes.POINTER(ctypes.c_int32), # pointer to buffer where result is written
+            ctypes.POINTER(ctypes.c_ulonglong) # pointer to size of value buffer (number of elements)
+        ]
+        self._nanopositionersdll.SA_CTL_GetProperty_i32.restype = ctypes.c_uint32   # result status
+
+        self._nanopositionersdll.SA_CTL_GetProperty_i64.argtypes = [
+            ctypes.c_uint32,                    # device handle
+            ctypes.c_int8,                      # index of addressed device, module, or channel
+            ctypes.c_uint32,                    # property key
+            ctypes.POINTER(ctypes.c_int64),     # pointer to buffer where result is written
+            ctypes.POINTER(ctypes.c_ulonglong)  # pointer to size of value buffer (# of elements)
+        ]
+        self._nanopositionersdll.SA_CTL_GetProperty_i64.restype = ctypes.c_uint32   # result status
+
+        # Settting device properties
+        self._nanopositionersdll.SA_CTL_SetProperty_i32.argtypes = [
+            ctypes.c_uint32,    # device handle
+            ctypes.c_int8,      # index of addressed device, module, or channel
+            ctypes.c_uint32,    # property key
+            ctypes.c_int32,     # value to write
+        ]
+        self._nanopositionersdll.SA_CTL_SetProperty_i32.restype = ctypes.c_uint32   # result status
+
+        self._nanopositionersdll.SA_CTL_SetProperty_i64.argtypes = [
+            ctypes.c_uint32,    # device handle
+            ctypes.c_int8,      # index of addressed device, module, or channel
+            ctypes.c_uint32,    # property key
+            ctypes.c_int64      # value to write
+        ]
+        self._nanopositionersdll.SA_CTL_SetProperty_i64.restype = ctypes.c_uint32   # result status
+
+        # Movement
+        self._nanopositionersdll.SA_CTL_Move.argtypes = [
+            ctypes.c_uint32,    # device handle
+            ctypes.c_int8,      # index of addressed device, module, or channel
+            ctypes.c_int64,     # move value
+            ctypes.c_uint32     # transmit handle
+        ]
+        self._nanopositionersdll.SA_CTL_Move.restype = ctypes.c_uint32  # result status
+
+        self._nanopositionersdll.SA_CTL_Stop.argtypes = [
+            ctypes.c_uint32,    # device handle
+            ctypes.c_int8,      # index of addressed device, module, or channel
+            ctypes.c_uint32     # transmit handle
+        ]
+        self._nanopositionersdll.SA_CTL_Stop.restype = ctypes.c_uint32  # result status
```

Comment:
Is there a way to read out the capacitance of the piezos? Could maybe be useful as an additional sanity check for the future.